From 2ca2906fe6db6a197932094f9cccb5e46a96421b Mon Sep 17 00:00:00 2001
From: Xiaoyu Yao <xyao@apache.org>
Date: Mon, 27 Jul 2015 07:28:41 -0700
Subject: [PATCH 1166/2748] HDFS-8785. TestDistributedFileSystem is failing in
 trunk. Contributed by Xiaoyu Yao.

(cherry picked from commit 2196e39e142b0f8d1944805db2bfacd4e3244625)
(cherry picked from commit 08616b3654d400c02d75bd08cd836d6a441df750)

Change-Id: I67b45ce5dd042b01310f3e391aeb1b9cd03b5c0e
---
 .../hadoop/hdfs/TestDistributedFileSystem.java     |    9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/TestDistributedFileSystem.java b/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/TestDistributedFileSystem.java
index bc00a55..ecb7e3a 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/TestDistributedFileSystem.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/TestDistributedFileSystem.java
@@ -1104,11 +1104,14 @@ public void testDFSClientPeerWriteTimeout() throws IOException {
       try {
         byte[] buf = new byte[10 * 1024 * 1024];
         peer.getOutputStream().write(buf);
-        Assert.fail("write should timeout");
+        long delta = Time.now() - start;
+        Assert.fail("write finish in " + delta + " ms" + "but should timedout");
       } catch (SocketTimeoutException ste) {
         long delta = Time.now() - start;
-        Assert.assertTrue("write timedout too soon", delta >= timeout * 0.9);
-        Assert.assertTrue("write timedout too late", delta <= timeout * 1.1);
+        Assert.assertTrue("write timedout too soon in " + delta + " ms",
+            delta >= timeout * 0.9);
+        Assert.assertTrue("write timedout too late in " + delta + " ms",
+            delta <= timeout * 1.2);
       } catch (Throwable t) {
         Assert.fail("wrong exception:" + t);
       }
-- 
1.7.9.5

